dbcon/joblist/resourcemanager.cpp

```diff
@@ -398,6 +398,14 @@ bool ResourceManager::getMemory(int64_t amount, bool patience)
   return ret1;
 }
 
+bool ResourceManager::getMemoryForce(int64_t amount, boost::shared_ptr<int64_t>& sessionLimit)
+{
+  bool ret1 = (atomicops::atomicSubRef(totalUmMemLimit, amount) >= 0);
+  bool ret2 = sessionLimit ? (atomicops::atomicSub(sessionLimit.get(), amount) >= 0) : ret1;
+
+  return (ret1 && ret2);
```

Comment:
is it correct, that `getMemoryForce` doesn't restore counters on failure?

Reply:
That's the point. We always want to process incoming RowGroups regardless of whether there is enough memory (so as not to lose data). In order to avoid being killed by the OOM-killer (for example, in a parallel query), I want this memory to be accounted (it will be returned later, either after destroying the TupleJoiner or converting it to DiskJoin).